lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -91,9 +91,8 @@ impl RegionInternal for MmapRegion {
             .pop()
             .ok_or(Error::RegionFull(self.capacity))?;
 
-        if slot.heap as usize % host_page_size() != 0 {
-            lucet_bail!("heap is not page-aligned; this is a bug");
-        }
+        slot.is_heap_page_aligned()
+            .expect("Heap must be page aligned");
```

Comment:
I don't think the original `Result`-oriented code is worth preserving. A `Slot` should never have any fields that aren't page-aligned.

```suggestion
        assert_neq!(slot.heap as usize % host_page_size(), 0, "heap must be page-aligned");
```